pyartifactory/objects.py

```diff
@@ -814,6 +849,25 @@ def download(self, artifact_path: str, local_directory_path: str = None) -> str:
         logger.debug("Artifact %s successfully downloaded", local_filename)
         return local_file_full_path
 
+    def download(self, artifact_path: str, local_directory_path: str = ".") -> str:
+        """
+        Download artifact (file or directory) into local directory.
+        :param artifact_path: Path to file or directory in Artifactory
+        :param local_directory_path: Local path to where the artifact will be downloaded
+        :return: File name
+        """
+        artifact_path = artifact_path.rstrip("/")
+        basename = artifact_path.split("/")[-1]
+        prefix = artifact_path.rsplit("/", 1)[0] + "/" if "/" in artifact_path else ""
+        for art in self._walk(artifact_path):
+            full_path = art.repo + art.path
+            local_path = join(local_directory_path, full_path[len(prefix) :])
+            if isinstance(art, ArtifactFolderInfoResponse):
+                os.mkdir(local_path)
```

Comment:
Would it be safer to use `os.makedirs(local_path, exist_ok=True)`? That way we can keep downloading in the same folder every time.

Reply:
I'm not sure I understand. The walk's `topdown` flag guarantees that missing parent directories will be created first, so I don't see where `os.mkdir()` would fail where `os.makedirs()` wouldn't. Or at least, if it would fail, it would already fail  prior to my PR, so I didn't add any implicit pre-conditions. :-)

Reply:
The issue isn't about non-existing folders, it's about already existing ones ^^
If you download a folder again for example, the folder structure is already there and `os.mkdir` errors if the directory already exists.

Reply:
Oh, indeed, good point ! :-)